src/browser/modules/Sidebar/ProjectFiles.tsx

```diff
@@ -0,0 +1,168 @@
+ */
+import React, { useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import { withBus } from 'react-suber'
+import { Bus } from 'suber'
+import { useMutation } from '@apollo/client'
+
+import Render from 'browser-components/Render'
+import { Drawer, DrawerHeader } from 'browser-components/drawer'
+import ProjectFilesScripts, { ProjectFilesError } from './ProjectsFilesScripts'
+import {
+  StyledSetting,
+  StyledSettingLabel,
+  StyledSettingTextInput
+} from './styled'
+import {
+  setProjectFileDefaultFileName,
+  updateCacheAddProjectFile
+} from './project-files.utils'
+import { SET_CONTENT } from 'shared/modules/editor/editorDuck'
+import { CYPHER_FILE_EXTENSION } from 'shared/services/export-favorites'
+import {
+  SELECT_PROJECT_FILE,
+  SAVE_PROJECT_FILE,
+  ADD_PROJECT_FILE,
+  PROJECT_FILES_MOUNTED,
+  PROJECT_FILES_UNMOUNTED
+} from './project-files.constants'
+
+interface IProjectFiles {
+  bus: Bus
+}
+
+const ProjectFiles = ({ bus }: IProjectFiles) => {
+  const [addFile, { error: apolloError }] = useMutation(ADD_PROJECT_FILE)
+  const [isSaveMode, setSaveMode] = useState(false)
+  const [fileName, setFileName] = useState('')
+  const [fileContents, setFileContents] = useState('')
+  const [error, setError] = useState('')
+
+  useEffect(() => {
+    let isStillMounted = true
+    // when ProjectFileButton is clicked in save mode
+    bus &&
+      bus.take(SAVE_PROJECT_FILE, editorValue => {
+        if (isStillMounted) {
+          setSaveMode(true)
+          setFileName(setProjectFileDefaultFileName(editorValue))
+          setFileContents(editorValue)
+          setError('')
+        }
+      })
+    // when a non-Project File action sets content in the editor
+    bus &&
+      bus.take(SET_CONTENT, () => {
+        if (isStillMounted) {
+          setSaveMode(false)
+          setFileName('')
+          setFileContents('')
+          setError('')
+        }
+      })
+    return () => {
+      isStillMounted = false
+    }
+  }, [])
+
+  useEffect(() => {
+    // send mounted/unmounted message
+    bus && bus.send(PROJECT_FILES_MOUNTED, '')
+    return () => bus && bus.send(PROJECT_FILES_UNMOUNTED, '')
+  }, [])
+
+  return (
+    <Drawer id="ProjectFiles">
+      <DrawerHeader>Project Files</DrawerHeader>
+      <Render if={isSaveMode}>
+        <StyledSetting>
+          <StyledSettingLabel title={'Enter a file name'}>
+            {'File Name'}
+            <StyledSettingTextInput
+              onChange={event => {
+                setFileName(event.target.value)
+              }}
+              value={fileName}
+            />
+          </StyledSettingLabel>
+        </StyledSetting>
+        <a
+          onClick={async () => {
+            if (!fileName.length) {
+              setError('File name cannot be empty')
+              return
+            }
+            if (fileName.includes('/') || fileName.includes('\\')) {
```

Comment:
Why not \\? Is a single backslash not problematic on windows? (don't use windows myself so not sure)

Reply:
it is searching for a single backslash to my knowledge - the `\` is escaping, so I had to do `\\`  for literal backslash.

Reply:
Oh right, the single backslack would escape the string char. I was thinking it didn't need escaping because it was in a string